src/OrbitClientData/CallstackData.cpp

```diff
@@ -105,23 +105,37 @@ std::vector<CallstackEvent> CallstackData::GetCallstackEventsOfTidInTimeRange(
 void CallstackData::ForEachCallstackEvent(
     const std::function<void(const orbit_client_protos::CallstackEvent&)>& action) const {
   std::lock_guard lock(mutex_);
-  for (const auto& tid_and_events : callstack_events_by_tid_) {
-    for (const auto& time_and_event : tid_and_events.second) {
-      action(time_and_event.second);
+  for (const auto& [unused_tid, events] : callstack_events_by_tid_) {
+    for (const auto& [unused_timestamp, event] : events) {
+      action(event);
+    }
+  }
+}
+
+void CallstackData::ForEachCallstackEventInTimeRange(
+    uint64_t min_timestamp, uint64_t max_timestamp,
+    const std::function<void(const orbit_client_protos::CallstackEvent&)>& action) const {
+  std::lock_guard lock(mutex_);
+  for (const auto& [unused_tid, events] : callstack_events_by_tid_) {
+    for (auto event_it = events.lower_bound(min_timestamp);
+         event_it != events.upper_bound(max_timestamp); ++event_it) {
```

Comment:
If max_timestamp is smaller than min_timestamp this could potentially be an endless loop. Maybe we should add a check here? Or use `std::min(min_timestamp, max_timestamp)` and `std::max(min_timestamp, max_timestamp)`?